src/algebra/lie/from_cartan_matrix.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2021 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import algebra.lie.free
+import algebra.lie.quotient
+
+/-!
+# Lie algebras from Cartan matrices
+
+Split semi-simple Lie algebras are uniquely determined by their Cartan matrix. Indeed, if `A` is
+an `l × l` Cartan matrix, the corresponding Lie algebra may be obtained as the Lie algebra on
+`3l` generators: $H_1, H_2, \ldots H_l, E_1, E_2, \ldots, E_l, F_1, F_2, \ldots, F_l$
+subject to the following relations:
+$$
+\begin{align}
+  [H_i, H_j] &= 0\\
+  [E_i, F_i] &= H_i \quad\mbox{ and }\quad [E_i, F_j] = 0 \quad\mbox{if $i ≠ j$}\\
+  [H_i, E_j] &= A_{ij}E_j\\
+  [H_i, F_j] &= -A_{ij}F_j\\
+  ad(E_i)^{1 - A_{ij}}(E_j) &= 0 \quad\mbox{if $i \ne j$}\\
+  ad(F_i)^{1 - A_{ij}}(F_j) &= 0 \quad\mbox{if $i \ne j$}\\
+\end{align}
+$$
+
+In this file we provide the above construction. It is defined for any matrix of integers but the
+results for non-Cartan matrices should be regarded as junk.
+
+## Main definitions
+
+  * `lie_algebra.from_cartan_matrix`
+
+## Tags
+
+lie algebra, semi-simple, cartan matrix
+-/
+
+universes u v w
+
+noncomputable theory
+
+variables (R : Type u) {B : Type v} [comm_ring R] [decidable_eq B] [fintype B]
+variables (cartan_matrix : matrix B B ℤ)
+
+namespace lie_algebra
+
+namespace from_cartan_matrix
```

Comment:
I think I would just call the namespace `cartan_matrix`. And maybe even outside `lie_algebra`.

Reply:
Done.